controller/subscriptionsproxycontroller.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * ownCloud - Calendar App
+ *
+ * @author Georg Ehrke
+ * @copyright 2016 Georg Ehrke <oc.list@georgehrke.com>
+ * @author Raghu Nayyar
+ * @copyright 2016 Raghu Nayyar <beingminimal@gmail.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+namespace OCA\Calendar\Controller;
+
+use OC\AppFramework\Http;
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\DataDisplayResponse;
+use OCP\Files\File;
+use OCP\Files\IRootFolder;
+use OCP\Files\NotFoundException;
+use OCP\IRequest;
+use OCP\IUserSession;
+
+class SubscriptionsProxyController extends Controller {
+
+	/**
+	 * @var IUserSession
+	 */
+	private $userSession;
+
+	/**
+	 * @var \OCP\Files\Folder
+	 */
+	private $storage;
+
+	/**
+	 * @param string $appName
+	 * @param IRequest $request an instance of the request
+	 * @param IUserSession $userSession
+	 * @param IRootFolder $storage
+	 */
+	public function __construct($appName, IRequest $request,
+									IUserSession $userSession, IRootFolder $storage) {
+		parent::__construct($appName, $request);
+		$this->userSession = $userSession;
+		$this->storage = $storage;
+	}
+
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 *
+	 * @param string $icsurl
+	 * @return DataDisplayResponse
+	 */
+	public function getIcsFile($icsurl) {
+		try {
+			$file = $this->storage->get('/' . basename($icsurl));
```

Comment:
>  It allows users to download any file (whatever size it has) and pollute the server's storage.

In order to fix that I can do a first HEAD request to get the file size and check it, then check it after it has been downloaded too. If the file is really too big (fake headers) then the handling will probably timeout anyhow.

> If we really want to cache it on the server, this should be done in the dav app

The CalDAV specification says subscriptions are only a list of remote urls, there's no handling of the files, therefore it would be only for this app.